app/controllers/roles_controller.rb

```diff
@@ -18,7 +20,15 @@ def show
 
   def load_from_graphql
     @role = Graphql::Role.find!(request.path)
-    @role.content_item
+    if @role.content_item.to_hash.key?("errors")
+      set_prometheus_labels("contains_errors" => true)
+      load_from_content_store
+    else
+      @role.content_item
+    end
+  rescue GdsApi::HTTPErrorResponse => e
+    set_prometheus_labels("graphql_status_code" => e.code)
+    load_from_content_store
```

Comment:
Given the change here, what is the current behaviour (ie before this PR)? Is it just failing if it gets an HTTPErrorResponse?

Reply:
Yes, as far as I understand